frameworks/supervised/exec.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+import tempfile as tmp
+import pandas as pd
+import numpy as np
+
+import sklearn
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+
+from frameworks.shared.callee import call_run, result, Timer
+
+log = logging.getLogger(os.path.basename(__file__))
+
+import supervised
+from supervised.automl import AutoML
+import shutil
+
+def run(dataset, config):
+    log.info("\n**** mljar-supervised ****\n")
+
+    is_classification = config.type == "classification"
+
+    X_train, X_test = dataset.train.X, dataset.test.X
+    y_train, y_test = dataset.train.y, dataset.test.y
+
+    X_train = pd.DataFrame(X_train, columns=[f"f{i}" for i in range(X_train.shape[1])])
+    X_test = pd.DataFrame(X_test, columns=[f"f{i}" for i in range(X_test.shape[1])])
+
+    y_train = y_train.flatten()
+    y_test = y_test.flatten()
+
+
+    # cast columns to have float types
+    for col in X_train.columns:
+        try: 
+            x1 = X_train[col].astype(float)
+            X_train[col] = x1
+            x2 = X_test[col].astype(float)
+            X_test[col] = x2
+        except Exception as e:
+            pass
+
+    print(
+        "We completely ignore the advice to optimize towards metric: {}.".format(
+            config.metric
+        )
+    )
+    
+    ml_task = None  # the AutoML will guess about ML task
+    # however, in case of multiclass classification, we will set the ml_task manually.
+    # AutoML assumes that multiclass calssification is when there is from 2 to 20 unique values.
+    # In this benchmark, there can be casses when it is much more classes than 20. That's why we set manually.
+    if is_classification and len(np.unique(y_train)) > 2:
+        ml_task = "multiclass_classification"
+
+    algorithms = ["Baseline", "Decision Tree", "Random Forest", "Xgboost", "LightGBM"]
+    if config.max_runtime_seconds <= 60:
+        algorithms = ["Baseline"]
+    results_path = "AutoML_results"
```

Comment:
is it relative to pwd?
can you please ensure it goes with the other outputs from this framework? 
Also when running multiple tasks/folds on the same instance, we usually want to distinguish all those outputs, that's why for other frameworks, we use the `make_subdir` function below:
```
results_path = make_subdir("results", config)
...

def make_subdir(name, config):
    subdir = os.path.join(config.output_dir, name, config.name, str(config.fold))
    touch(subdir, as_dir=True)
    return subdir
```

Reply:
I've used `make_subdir` method.